velox/functions/prestosql/tests/MapFromEntriesTest.cpp

```diff
@@ -0,0 +1,202 @@
+       variant::row({"yellow", "yellow rose flowers"})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<StringView, StringView>(
+      {{{"red"_sv, "red shiny car ahead"_sv},
+        {"blue"_sv, "blue clear sky above"_sv},
+        {"yellow"_sv, "yellow rose flowers"_sv}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, dateKeyAndBigintValue) {
+  auto rowType = ROW({DATE(), BIGINT()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({Date(0), (int64_t)6945148781159939835}),
+       variant::row({Date(10), (int64_t)6945148781159939835}),
+       variant::row({Date(20), (int64_t)6945148781159939835})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<Date, int64_t>(
+      {{{Date(0), 6945148781159939835},
+        {Date(10), 6945148781159939835},
+        {Date(20), 6945148781159939835}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, nullMapEntries) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant(TypeKind::ROW), variant::row({1, 11})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  VELOX_ASSERT_THROW(
+      evaluateExpr("map_from_entries(C0)", {input}),
+      "map entry cannot be null");
+  EXPECT_NO_THROW(evaluateExpr("try(map_from_entries(C0))", {input}));
+}
+
+TEST_F(MapFromEntriesTest, nullKeys) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({variant::null(TypeKind::INTEGER), 0}),
+       variant::row({1, 11})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  VELOX_ASSERT_THROW(
+      evaluateExpr("map_from_entries(C0)", {input}), "map key cannot be null");
+  EXPECT_NO_THROW(evaluateExpr("try(map_from_entries(C0))", {input}));
+}
+
+TEST_F(MapFromEntriesTest, duplicateKeys) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({1, 10}), variant::row({1, 11}), variant::row({2, 22})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  VELOX_ASSERT_THROW(
+      evaluateExpr("map_from_entries(C0)", {input}),
+      "Duplicate map keys (1) are not allowed");
+  EXPECT_NO_THROW(evaluateExpr("try(map_from_entries(C0))", {input}));
+}
+
+TEST_F(MapFromEntriesTest, nullValues) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({1, variant::null(TypeKind::INTEGER)}),
+       variant::row({2, 22}),
+       variant::row({3, 33})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected =
+      makeMapVector<int32_t, int32_t>({{{1, std::nullopt}, {2, 22}, {3, 33}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, moreThanTwoColumnsInRow) {
```

Comment:
This test case is not necessary.

Reply:
Why do you think this edge case is unnecessary? Is it guarded by the function `signature()` definition?